components/provider_ppucd/src/parse_ppucd.rs

```diff
@@ -447,4 +652,472 @@ mod gen_properties_test {
+
+        let jt_u = UnicodeProperty {
+            name: Cow::Borrowed("11=5"),
+            inv_list: vec![5888, 5901, 5902, 5906],
+        };
+        exp_uni_props_set.insert(jt_u);
+        let jt_t = UnicodeProperty {
+            name: Cow::Borrowed("11=4"),
+            inv_list: vec![5906, 5909],
+        };
+        exp_uni_props_set.insert(jt_t);
+        let alpha = UnicodeProperty {
+            name: Cow::Borrowed("Alpha"),
+            inv_list: vec![5888, 5901, 5902, 5908],
+        };
+        exp_uni_props_set.insert(alpha);
+        let nfd_qc_y = UnicodeProperty {
+            name: Cow::Borrowed("15=1"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(nfd_qc_y);
+        let nfc_qc_y = UnicodeProperty {
+            name: Cow::Borrowed("14=2"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(nfc_qc_y);
+        let nfkd_qc_y = UnicodeProperty {
+            name: Cow::Borrowed("17=1"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(nfkd_qc_y);
+        let nfkc_qc_y = UnicodeProperty {
+            name: Cow::Borrowed("16=2"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(nfkc_qc_y);
+        let jg_no_joining_group = UnicodeProperty {
+            name: Cow::Borrowed("10=71"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(jg_no_joining_group);
+        let dt_none = UnicodeProperty {
+            name: Cow::Borrowed("3=11"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(dt_none);
+        let nt_none = UnicodeProperty {
+            name: Cow::Borrowed("18=2"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(nt_none);
+        let ea_n = UnicodeProperty {
+            name: Cow::Borrowed("4=3"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(ea_n);
+        let vo_r = UnicodeProperty {
+            name: Cow::Borrowed("21=0"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(vo_r);
+        let ccc_9 = UnicodeProperty {
+            name: Cow::Borrowed("2=9"),
+            inv_list: vec![5908, 5909],
+        };
+        exp_uni_props_set.insert(ccc_9);
+        let hst_na = UnicodeProperty {
+            name: Cow::Borrowed("7=3"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
+        exp_uni_props_set.insert(hst_na);
+        let gr_link = UnicodeProperty {
+            name: Cow::Borrowed("Gr_Link"),
+            inv_list: vec![5908, 5909],
+        };
+        exp_uni_props_set.insert(gr_link);
+        let bpt_n = UnicodeProperty {
+            name: Cow::Borrowed("1=1"),
+            inv_list: vec![5888, 5901, 5902, 5909],
+        };
```

Comment:
`option`: for maintenance convenience reasons, I often use `From` for scenarios like this to end up with:

```rust
let bpt_n = ("1=1", vec![5888, 5901, 5902, 5909]).into();
```

or, in case of tests even:

```rust
let samples = &[
    ("1=1", vec![5888, 5901, 5902, 5909]),
    ...
];
for elem in samples {
  test(elem.into());
}
```

Reply:
I held off on this because I find the syntax for defining structs is more readable -- it includes the field names before the instance value => named arguments => not position-based like a tuple. And therefore more maintainable. Since named arguments are simpler, I'm okay with the the verbosity tradeoff.